src/OpenTelemetry.Api/Trace/SpanAttributes.cs

```diff
@@ -0,0 +1,129 @@
+﻿// <copyright file="SpanAttributes.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// A class that represents the span attributes. Read more here https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/common/common.md#attributes.
+    /// </summary>
+    public class SpanAttributes
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpanAttributes"/> class.
+        /// </summary>
+        /// <param name="initialAttributes">Initial attributes for the <see cref="SpanAttributes"/>.</param>
+        public SpanAttributes(IEnumerable<KeyValuePair<string, object>> initialAttributes)
```

Comment:
lets provide constructor with no args too.

Reply:
with @alanwest suggestions, we would have the default constructor + method to AddRange. That would simplify things. Or should we maintain the constructor with initialAttributes as well?